lib/src/kbasesearchengine/KBaseSearchEngineServer.java

```diff
@@ -212,15 +212,18 @@ public void timeStat(GUID guid, long loadMs, long parseMs, long indexMs) {
      * @param   params   instance of type {@link kbasesearchengine.SearchTypesInput SearchTypesInput}
      * @return   instance of type {@link kbasesearchengine.SearchTypesOutput SearchTypesOutput}
      */
-    @JsonServerMethod(rpc = "KBaseSearchEngine.search_types", async=true)
+    @JsonServerMethod(rpc = "KBaseSearchEngine.search_types", authOptional=true, async=true)
     public SearchTypesOutput searchTypes(SearchTypesInput params, AuthToken authPart, RpcContext jsonRpcContext) throws Exception {
         SearchTypesOutput returnVal = null;
         //BEGIN search_types
-        returnVal = search.searchTypes(params, authPart.getUserName());
+        returnVal = search.searchTypes(params, getUserName(authPart));
         //END search_types
         return returnVal;
     }
 
+    private String getUserName(final AuthToken authPart){
```

Comment:
This will be lost the next time the server is recompiled. Remember all code has to go between //BEGIN and //END markers.

Reply:
what then how do I do helper functions